gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobLauncher.java

```diff
@@ -269,6 +271,62 @@ protected void executeCancellation() {
     }
   }
 
+  protected void removeTaskFromRunningHelixJob(List<String> workUnitIdsToRemove) throws IOException {
+    String jobName = this.jobContext.getJobId();
+    try (ParallelRunner stateSerDeRunner = new ParallelRunner(this.stateSerDeRunnerThreads, this.fs)) {
+      for (String workUnitId : workUnitIdsToRemove) {
+        int helixOperationRetryTime = 3;
+        while (helixOperationRetryTime > 0) {
+          String taskId = workUnitToHelixConfig.get(workUnitId).getId();
+          boolean remove = HelixUtils.deleteTaskFromHelixJob(this.helixWorkFlowName, jobName, taskId, this.helixTaskDriver);
+          if (remove) {
+            log.info(String.format("Removed helix task %s with gobblin task id  %s from helix job %s:%s ", taskId,
+                workUnitId, this.helixWorkFlowName, jobName));
+            deleteWorkUnitFromStateStore(workUnitId, stateSerDeRunner);
+            log.info(String.format("remove task state for %s in state store", workUnitId));
+            this.workUnitToHelixConfig.remove(workUnitId);
+          break;
+          } else {
+            helixOperationRetryTime--;
+            if (helixOperationRetryTime == 0) {
+              log.error(String.format("Failed to remove task %s from helix job %s:%s ", workUnitId, this.helixWorkFlowName,
+                  jobName));
+              throw new IOException(
+                  String.format("Cannot remove task %s from helix job %s:%s after 3 times", workUnitId, this.helixWorkFlowName, jobName));
+            }
+          }
+        }
+      }
+    }
+  }
+
+  protected void addTaskToRunningHelixJob(List<WorkUnit> workUnitsToAdd) throws IOException {
```

Comment:
there's a lot of logic duplication between these two method. What they do seem to be as simple as: 
```
for workunit: workunits: 
    helixTask = getHelixTask (workunit) 
   retryer.execute(helixTask, helixOps)
```

Can you try to refactor them a bit ? There's retry object you could use, or create your own and pass-in the lambda expression